lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -254,11 +310,124 @@ describe('BasicLtiLaunchApp', () => {
     });
   });
 
+  describe('concurrent fetching of groups and content', () => {
+    let contentUrlResolve,
+      contentUrlReject,
+      groupsCallResolve,
+      groupsCallReject;
+
+    beforeEach(() => {
+      // Will attempt to fetch the 1. content url and 2. groups.
+      fakeConfig.api = {
+        authToken: 'dummyAuthToken',
+        viaCallbackUrl: 'https://lms.hypothes.is/api/files/1234',
+        sync: {
+          data: {
+            course: {
+              context_id: '12345',
+              custom_canvas_course_id: '101',
+            },
+          },
+          path: '/api/sync',
+        },
+      };
+      fakeApiCall
+        .withArgs({
+          authToken: 'dummyAuthToken',
+          path: 'https://lms.hypothes.is/api/files/1234',
+        })
+        .returns(
+          new Promise((resolve, reject) => {
+            contentUrlResolve = resolve;
+            contentUrlReject = reject;
+          })
+        );
+
+      fakeApiCall
+        .withArgs({
+          authToken: 'dummyAuthToken',
+          path: '/api/sync',
+          data: {
+            course: {
+              context_id: '12345',
+              custom_canvas_course_id: '101',
+            },
+          },
+        })
+        .returns(
+          new Promise((resolve, reject) => {
+            groupsCallResolve = resolve;
+            groupsCallReject = reject;
+          })
+        );
+    });
+
+    it('renders the spinner until both groups and contentUrl requests finish', async () => {
+      const wrapper = renderLtiLaunchApp();
+
+      // Spinner should not go away after first request
+      contentUrlResolve({
+        via_url: 'https://via.hypothes.is/123',
+      });
+      assert.isTrue(wrapper.find('Spinner').exists());
```

Comment:
I wonder if would have a similar effect?

`assert.rejects(await waitForSpinnerToBeRemoved(...))`

Reply:
As long as you set a very short timeout (say 1ms) on the `waitForXXX` call, and the `waitForXXX` call is calling `wrapper.update()` before each. check, then yes this could work.

Reply:
(If you don't set a short timeout, `waitForXXX` will of course just hang for several seconds waiting for something that doesn't happen).